geps/gep-3539/index.md

```diff
@@ -0,0 +1,200 @@
+By default, IP address(es) from a pool specified by a CIDR block will be assigned unless a static IP is 
+configured in the _addresses_ field as shown above. The CIDR block may be configured using a custom CR. 
+Subject to further discussion, it may make sense to have a GatewayCIDR resource available upstream to 
+specify an IP address range for Gateway IP allocation.
+
+Finally the specific Route and EndpointSelector resources must be created in order to set up the backend 
+pods for the configured ClusterIP.
+```yaml
+{% include 'standard/clusterip-gateway/customroute.yaml' %}
+```
+
+### Backends on Listeners
+
+As seen above, Gateway API requires at least three CRs to be defined. This introduces some complexity. 
+GEP-1713 proposes the addition of a ListenerSet resource to allow sets of listeners to attach to a Gateway. 
+As a part of discussions around this topic, the idea of directly adding backendRefs to listeners has come 
+up. Allowing backendRefs directly on the listeners eliminates the need to have Route objects for simple 
+cases. More complex traffic splitting and advanced load balancing cases can still use Route attachments via 
+allowedRoutes. 
+
+### DNS
+
+ClusterIP Gateways in the cluster need to have consistent DNS names assigned to allow ClusterIP lookup by 
+name rather than IP address. DNS A and/or AAAA record creation needs to happen when Kubernetes publishes 
+information about Gateways, in a manner similar to ClusterIP Service creation behavior. DNS nameservers 
+in pods’ /etc/resolv.conf need to be programmed accordingly by kubelet.
+
+```
+<name of gateway>.<gateway-namespace>.gw.cluster.local
+```
+
+This results in the following search option entries in Pods’ /etc/resolv.conf:
+```
+search <ns>.gw.cluster.local  gw.cluster.local  cluster.local
+```
+
+### Cross-namespace References
+
+Gateway API allows for Routes in different namespaces to attach to the Gateway. 
+
+When modeling ClusterIP service networking, the simplest recommendation might be to keep Gateway and Routes 
+within the same namespace. While cross namespace routing would work and allow for evolved functionality, 
+it may make supporting certain cases tricky. One specific example for this case is the pod DNS resolution 
+support of the following format 
+
+```
+pod-ipv4-address.gateway-name.my-namespace.gw.cluster-domain.example
+```
+
+If Gateway and Routes (and hence the backing pods) are in different namespaces, there arises ambiguity in 
+whether and how to support this pod DNS resolution format.
+
+## LoadBalancer and NodePort Services
+
+Extending the concept further to LoadBalancer and NodePort type services follows a similar pattern. The idea 
+is to have a GatewayClass corresponding to each type of service networking behavior that needs to be modeled 
+and supported.
+
+![image displaying gatewayclasses to represent different service types](images/gatewayclasses-lb-np.png "image displaying gatewayclasses to represent different service types")
```

Comment:
image missing or incorrect file name?

Reply:
Was a missing image. Fixed. Thanks for catching